modules/crowdx/src/api/when.ts

```diff
@@ -0,0 +1,16 @@
+import { Observable } from './observable'
+import { observe } from './index'
+
+type WhenCondition<ValueT> = (value: ValueT) => boolean
+
+const when = async <ValueT>(num: Observable<ValueT>, condition: WhenCondition<ValueT>): Promise<ValueT> => {
+  return await new Promise<ValueT>((resolve) => {
+    observe(num, (val) => {
+      if (condition(val as ValueT)) {
+        resolve(val as ValueT)
```

Comment:
Observe can receive the ValutT type parameter, so you wont need to do any typecasting later on